nengo/builder/builder.py

```diff
@@ -131,8 +132,15 @@ def build(self, obj, *args, **kwargs):
             The object to build into this model.
         """
         built = self.builder.build(self, obj, *args, **kwargs)
+
+        def NoneFunction(other):
+            return
+
+        function_to_call = NoneFunction
         if self.build_callback is not None:
-            self.build_callback(obj)
+            function_to_call = self.build_callback
+        if function_to_call is not NoneFunction:
+            function_to_call(obj)
```

Comment:
This seems much more confusing than the original code. What was `pylint`'s problem?

Also, as a general principle, I don't mind adding a `pylint: disable=...` to a particular line if it means keeping the code cleaner (sometimes, `pylint` is just out to lunch IMO). But I think having the disables on specific lines is still better than having them in the disable list, since then we're not ignoring them everywhere.